File: hw/tipantiza/u3/hw17OOPConcepts/LibrarySystem_js/model/book.js
class Book {
    constructor(id = null, title = null, author = null, isbn = null, publicationYear = null, category = null) {
        this._id = id;
        this._title = title;
        this._author = author;
        this._isbn = isbn;
        this._publicationYear = publicationYear;
        this._available = true;
        this._category = category;
    }
    
    get id() {
        return this._id;
    }
    
    set id(value) {
        this._id = value;
    }
    
    get title() {
        return this._title;
    }
    
    set title(value) {
        this._title = value;
    }
    
    get author() {
        return this._author;
    }
    
    set author(value) {
        this._author = value;
    }
    
    get isbn() {
        return this._isbn;
    }
    
    set isbn(value) {
        this._isbn = value;
    }
    
    get publicationYear() {
        return this._publicationYear;
    }
    
    set publicationYear(value) {
        this._publicationYear = value;
    }
    
    get available() {
        return this._available;
    }
    
    set available(value) {
        this._available = value;
    }
    
    get category() {
        return this._category;
    }
    
    set category(value) {
        this._category = value;
    }
    
    borrow() {
        if (!this._available) {
            throw new Error("Book is not available");
        }
        this._available = false;
    }
    
    returnBook() {
        this._available = true;
    }
    
    toString() {
        return `Book: ${this._title} - ${this._author} (${this._isbn})`;
    }
}

module.exports = Book;

File: hw/tipantiza/u3/hw17OOPConcepts/LibrarySystem/src/main/java/ec/edu/espe/librarysystem/model/Book.java

package ec.edu.espe.librarysystem.model;

/**
 *
 * @author Alexander Tipantiza, The Softwarrios, @ESPE
 */

public class Book {
    private String id;
    private String title;
    private String author;
    private String isbn;
    private int publicationYear;
    private boolean available;
    private String category;

    public Book() {
    }

    public Book(String id, String title, String author, String isbn, 
                int publicationYear, String category) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.isbn = isbn;
        this.publicationYear = publicationYear;
        this.available = true;
        this.category = category;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public int getPublicationYear() {
        return publicationYear;
    }

    public void setPublicationYear(int publicationYear) {
        this.publicationYear = publicationYear;
    }

    public boolean isAvailable() {
        return available;
    }

    public void setAvailable(boolean available) {
        this.available = available;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public void borrow() {
        if (!available) {
            throw new IllegalStateException("Book is not available");
        }
        this.available = false;
    }

    public void returnBook() {
        this.available = true;
    }

    @Override
    public String toString() {
        return "Book: " + title + " - " + author + " (" + isbn + ")";
    }
}
